PixelDriver.h

```diff
@@ -106,6 +106,10 @@ class PixelDriver {
     void ICACHE_RAM_ATTR show();
     uint8_t* getData();
 
+    inline bool hasWhite() {
+        return this->color_count == 4;
```

Comment:
As a general engineering rule, I recommend avoiding the overloading of meaning of a variable.
Instead, I would recommend storing this value separately, perhaps initializing the variable based on the configuration's attached LED type?

Reply:
Also, it's not clear if there's any adjustment available in this for the different white color temperatures.
What color temperature did you adjust the LED effects to support?
Generally, I would (at a minimum) recommend tracking the white temperature of each white pixel.

Also, expose this via the UI ... so it's clear that the effects were designed based on a given color 
temperature.  (e.g., 2700K?  5000K? ... it really affects the result...)

Reply:
So full disclosure, I am totally an amateur with this stuff. I have an RGBW strip that I wanted to drive with this code and so I figured out how to make it work and thought I would contribute it back to whomever it might benefit.

I also tried to avoid changing too much about how it currently works. `color_count` was already there, so I used it. Curious how you mean that it is overloading it's meaning though? The value is initialized in PixelDriver.cpp, line 95 based on the configuration.

I'll be honest I didn't spend too much time with the effects, except to modify them to set there is W pixel present.

Reply:
Also, isnt the color temperature fixed dependending on the specific LED strip? Like I don't think you can programmatically change their color. From what I understand there are RGBCW, RGBWW and RGBNW variants for cold white, warm white and natural white, but they are all driven by SK6812 the same.

Reply:
Hi Dennis,

<details><summary>Overloading of meaning</summary><P/>
This is not an industry-standard term, just prose.  Here are two articles that should help explain:
* [Separation of concerns](https://en.wikipedia.org/wiki/Separation_of_concerns)
* [Single Responsibility Principle](https://en.wikipedia.org/wiki/Single_responsibility_principle)

Thus, a single variable should correspond to a single concept.  The `color_count` indicates how many distinct LEDs per pixel need data.  This does not indicate a color ordering, nor does it indicate if any of the LEDs are white (nor the color temperature of white).

Separating it, there would be a method to uniquely identify, for each color, what color the LED in that pixel is.
</details>

<details><summary>Color Temperature Fixed?</summary><P/>

Yes, the color temperature for a given white LED is fixed at manufacture.
Yes, it's typically the same for all white LEDs in common pixel strips.
Yes, the protocol for (using your terms) RGBcW, RGBwW, RGBnW is identical, as it the driver chip (SK6812) ... it requires user input to know which color temperature, just like it needs user input to know the RGB color order.

</details>

In short, you should give yourself more credit!  You do appear to understand things quite well.  :)